docs/advanced/picking.md

```diff
@@ -51,42 +38,34 @@ system, which most layers use.
 To take full control of picking, a layer need to take the following steps:
 
 1. Add a picking color attribute during initialization:
-  ```js
-  initializeState() {
+```js
+initializeState() {
 
-    ...
+  ...
 
-    this.state.attributeManager.add({
-      pickingColors: {size: 3, type: GL.UNSIGNED_BYTE, update: this.calculatePickingColors}
-    });
-  }
-  ```
+  this.state.attributeManager.add({
+    pickingColors: {size: 3, type: GL.UNSIGNED_BYTE, update: this.calculatePickingColors}
+  });
+}
+```
 - Calculate the attribute by providing a different picking color for every object that
 you need to differentiate, such as:
-  ```js
-  calculatePickingColors(attribute) {
-    const {data} = this.props;
-    const {value, size} = attribute;
-    
-    let i = 0;
-    for (const object of data) {
-      const pickingColor = this.encodePickingColor(i);
-      value[i++] = pickingColor[0];
-      value[i++] = pickingColor[1];
-      value[i++] = pickingColor[2];
-    }
+```js
+calculatePickingColors(attribute) {
+  const {data} = this.props;
+  const {value, size} = attribute;
+
+  let i = 0;
+  for (const object of data) {
+    const pickingColor = this.encodePickingColor(i);
```

Comment:
> const pickingColor = this.encodePickingColor(i);

Nit: should we use a separate index to pass to encodePickingColor for clarity, may be like:
```
let i = 0;
let objectIndex = 0;
 for (const object of data) {
    const pickingColor = this.encodePickingColor(objectIndex++);
    value[i++] = ...
```